internal/websocket/options.go

```diff
@@ -179,22 +179,38 @@ func SendTimeout(d time.Duration) Option {
 		})
 }
 
-// HTTPClient is the HTTP client used for connection attempts.
-func HTTPClient(client *http.Client) Option {
+// HTTPClient is the configuration for the HTTP client used for connection attempts.
+func HTTPClient(c arrangehttp.ClientConfig) Option {
 	return optionFunc(
 		func(ws *Websocket) error {
-			var err error
-			if client == nil {
-				// Can't use http.DefaultClient since its Transport is nil.
-				client, err = arrangehttp.ClientConfig{}.NewClient()
+			// test new client
+			if _, err := c.NewClient(); err != nil {
+				return errors.Join(err, ErrMisconfiguredWS)
 			}
 
-			if err != nil {
-				return errors.Join(ErrMisconfiguredWS, err)
+			ws.httpClientConfig = c
+
+			return nil
+		})
+}
+
+// HTTPClientWithForceSets is the configuration for the HTTP client with recommended force sets, used for connection attempts.
+func HTTPClientWithForceSets(c arrangehttp.ClientConfig) Option {
```

Comment:
A new option that force sets whatever HTTPClient fields we absolutely need, allowing us to make most required configuration changes happen all at once.